type-definitions/ts-tests/set.ts

```diff
@@ -1,294 +1,262 @@
+import { expect, test } from 'tstyche';
 import { Set, Map, Collection } from 'immutable';
 
-{
-  // #constructor
+test('#constructor', () => {
+  expect(Set()).type.toEqual<Set<unknown>>();
 
-  // $ExpectType Set<unknown>
-  Set();
+  expect(Set<number>()).type.toEqual<Set<number>>();
 
-  const numberSet: Set<number> = Set<number>();
-  const numberOrStringSet: Set<number | string> = Set([1, 'a']);
+  expect(Set([1, 'a'])).type.toEqual<Set<number | string>>();
+});
 
-  // $ExpectError
-  const invalidNumberSet: Set<number> = Set([1, 'a']);
```

Comment:
Not sure these are good test (here and similarly in other files). Perhaps? It can be written like this:

```ts
expect<Set<number>>().type.not.toBeAssignable(Set([1, 'a']));
```

, but I don’s like how it reads. Not clear what is going on.

Here is an idea. I could add `.toBeAssignableTo()` and `.toBeAssignableFrom()` matchers. Those read well and keeping in mind that `.toEqual()` already exists these two sound similar to `.toBeGreaterThan()` and `.toBeLessThan()`.

And the assertion would look like this:

```ts
expect(Set([1, 'a'])).type.not.toBeAssignableTo<Set<number>>();
```

This looks better. Or?